jax/_src/numpy/fft.py

```diff
@@ -522,18 +522,131 @@ def irfft(a: ArrayLike, n: int | None = None,
   return _fft_core_1d('irfft', xla_client.FftType.IRFFT, a, n=n, axis=axis,
                       norm=norm)
 
-@implements(np.fft.hfft)
+
 def hfft(a: ArrayLike, n: int | None = None,
          axis: int = -1, norm: str | None = None) -> Array:
+  """Compute a 1-D FFT of an array whose spectrum has Hermitian symmetry.
+
+  JAX implementation of :func:`numpy.fft.hfft`.
+
+  Args:
+    a: input array.
+    n: optional, int. Specifies the dimension of the result along ``axis``. If
+      not specified, ``n = 2*(m-1)``, where ``m`` is the dimension of ``a``
+      along ``axis``.
+    axis: optional, int, default=-1. Specifies the axis along which the transform
+      is computed. If not specified, the transform is computed along axis -1.
+    norm: optional, string. The normalization mode. "backward", "ortho" and "forward"
+      are supported. Default is "backward".
+
+  Returns:
+    An array containing the one-dimensional discret Fourier transform of ``a``
+    by exploiting its inherent Hermitian-symmetry, having a dimension of ``n``
+    along ``axis``.
+
+  See also:
+    - :func:`jax.numpy.fft.ihfft`: Computes a one-dimensional inverse FFT of an
+      array whose spectrum has Hermitian symmetry.
+    - :func:`jax.numpy.fft.fft`: Computes a one-dimensional discrete Fourier
+      transform.
+    - :func:`jax.numpy.fft.rfft`: Computes a one-dimensional discrete Fourier
+      transform of a real-valued input.
+
+  Examples:
+    >>> x = jnp.array([[1, 3, 5, 7],
+    ...                [2, 4, 6, 8]])
+    >>> jnp.fft.hfft(x)
+    Array([[24., -8.,  0., -2.,  0., -8.],
+           [30., -8.,  0., -2.,  0., -8.]], dtype=float32)
+
+    This is equivalent to computing ``jnp.fft.fft`` of the following array ``x1``:
```

Comment:
Not quite equivalent because of the `dtype`s!

Reply:
Modified the statement as "This value is equal to the real component of the discrete Fourier transform
    of the following array ``x1`` computed using ``jnp.fft.fft``."